packages/js/e2e-utils-playwright/src/editor.js

```diff
@@ -0,0 +1,164 @@
+	);
+
+	if ( isWelcomeGuideActive ) {
+		await page.evaluate( () =>
+			wp.data.dispatch( 'core/edit-post' ).toggleFeature( 'welcomeGuide' )
+		);
+	}
+};
+
+export const openEditorSettings = async ( { page } ) => {
+	// Open Settings sidebar if closed
+	if ( await page.getByLabel( 'Editor Settings' ).isVisible() ) {
+		console.log( 'Editor Settings is open, skipping action.' );
+	} else {
+		await page.getByLabel( 'Settings', { exact: true } ).click();
+	}
+};
+
+export const getCanvas = async ( page ) => {
+	return page.frame( 'editor-canvas' ) || page;
+};
+
+export const goToPageEditor = async ( { page } ) => {
+	const responsePromise = page.waitForResponse(
+		( response ) =>
+			response.url().includes( '//page' ) && response.status() === 200
+	);
+	await page.goto( 'wp-admin/post-new.php?post_type=page' );
+	await disableWelcomeModal( { page } );
+	await closeChoosePatternModal( { page } );
+	await responsePromise;
+};
+
+export const goToPostEditor = async ( { page } ) => {
+	const responsePromise = page.waitForResponse(
+		( response ) =>
+			response.url().includes( '//single' ) && response.status() === 200
+	);
+	await page.goto( 'wp-admin/post-new.php' );
+	await disableWelcomeModal( { page } );
+	await responsePromise;
+};
+
+export const fillPageTitle = async ( page, title ) => {
+	await ( await getCanvas( page ) ).getByLabel( 'Add title' ).click();
+	await ( await getCanvas( page ) ).getByLabel( 'Add title' ).fill( title );
+};
+
+export const insertBlock = async ( page, blockName, wpVersion = null ) => {
+	await page
+		.getByRole( 'button', {
+			name: 'Toggle block inserter',
+			expanded: false,
+		} )
+		.click();
+	await page.getByPlaceholder( 'Search', { exact: true } ).fill( blockName );
+	await page.getByRole( 'option', { name: blockName, exact: true } ).click();
+
+	// In WP 6.6 'Toggle block inserter' button closes the inserter as expected,
+	// but trying to immediately open it again will fail in Playwright, while manually it works.
+	// We have tests that insert multiple blocks and fail because of this.
+	// Using the new 'Close block inserter' button added in WP 6.6 works fine.
+	if ( wpVersion && wpVersion <= 6.5 ) {
+		await page
+			.getByRole( 'button', {
+				name: 'Toggle block inserter',
+				expanded: true,
+			} )
+			.click();
+	} else {
+		await page
+			.getByRole( 'button', {
+				name: 'Close block inserter',
+			} )
+			.click();
+	}
+};
+
+export const insertBlockByShortcut = async ( page, blockName ) => {
+	const canvas = await getCanvas( page );
+	await canvas.getByRole( 'button', { name: 'Add default block' } ).click();
+	await canvas
+		.getByRole( 'document', {
+			name: 'Empty block; start writing or type forward slash to choose a block',
+		} )
+		.pressSequentially( `/${ blockName }` );
+	await expect(
+		page.getByRole( 'option', { name: blockName, exact: true } )
+	).toBeVisible();
+	await page.getByRole( 'option', { name: blockName, exact: true } ).click();
+	await expect(
```

Comment:
We probably shouldn't be putting assertions into utilities.  We should remove these and instead have the assertion separate.